src/core/connection.c

```diff
@@ -6101,6 +6388,20 @@ QuicConnProcessApiOperation(
         ApiCtx->CONN_START.ServerName = NULL;
         break;
 
+    case QUIC_API_TYPE_CONN_SEND_RESUMPTION_TICKET:
+        Status =
+            QuicConnSendResumptionTicket(
+                Connection,
+                ApiCtx->CONN_SEND_RESUMPTION_TICKET.AppDataLength,
+                ApiCtx->CONN_SEND_RESUMPTION_TICKET.ResumptionAppData);
+        ApiCtx->CONN_SEND_RESUMPTION_TICKET.ResumptionAppData = NULL;
+        if (ApiCtx->CONN_SEND_RESUMPTION_TICKET.Flags & QUIC_SEND_RESUMPTION_FLAG_FINAL) {
+            QUIC_DBG_ASSERT(QuicConnIsServer(Connection)); // TODO: handle client-side
+            Connection->State.ResumptionEnabled = FALSE;
+            QuicConnCleanupServerResumptionState(Connection); // BUG: With Async processing, this frees the memory before it's used.
```

Comment:
I think we should do synchronous processing. Always require the data be ready to query right after.

Reply:
Agreed. I'll wire that up in the PR with the TLS changes.